src/masternodes/rpc_icxorderbook.cpp

```diff
@@ -0,0 +1,1443 @@
+                            {"", RPCArg::Type::OBJ, RPCArg::Optional::OMITTED, "",
+                                {
+                                    {"txid", RPCArg::Type::STR_HEX, RPCArg::Optional::NO, "The transaction id"},
+                                    {"vout", RPCArg::Type::NUM, RPCArg::Optional::NO, "The output number"},
+                                },
+                            },
+                        },
+                    },
+                },
+                RPCResult{
+                        "\"hash\"                  (string) The hex-encoded hash of broadcasted transaction\n"
+                },
+                RPCExamples{
+                        HelpExampleCli("icx_makeoffer", "'{\"orderTx\":\"<txid>\",\"amount\":\"10\","
+                                                        "\receiveAddress\":\"<address>\",}'")
+                        + HelpExampleCli("icx_makeoffer", "'{\"orderTx\":\"txid\",\"amount\":\"10\","
+                                                        "\"ownerAddress\":\"<address>\",\"receivePubkey\":\"<pubkey>\"}'")
+                },
+     }.Check(request);
+
+    if (pwallet->chain().isInitialBlockDownload())
+        throw JSONRPCError(RPC_CLIENT_IN_INITIAL_DOWNLOAD, "Cannot make offer while still in Initial Block Download");
+
+    pwallet->BlockUntilSyncedToCurrentChain();
+    LockedCoinsScopedGuard lcGuard(pwallet);
+
+    RPCTypeCheck(request.params, {UniValue::VOBJ}, false);
+    if (request.params[0].isNull()) {
+        throw JSONRPCError(RPC_INVALID_PARAMETER,
+                           "Invalid parameters, arguments 1 must be non-null and expected as object at least with "
+                           "{\"orderTx\",\"amount\", \"receivePubkey|receiveAddress\"}");
+    }
+    UniValue metaObj = request.params[0].get_obj();
+    UniValue const & txInputs = request.params[1];
+
+    CICXMakeOffer makeoffer;
+
+    if (!metaObj["orderTx"].isNull())
+        makeoffer.orderTx = uint256S(metaObj["orderTx"].getValStr());
+    else
+        throw JSONRPCError(RPC_INVALID_PARAMETER,"Invalid parameters, argument \"orderTx\" must be non-null");
+    if (!metaObj["amount"].isNull())
+        makeoffer.amount = AmountFromValue(metaObj["amount"]);
+    else 
+        throw JSONRPCError(RPC_INVALID_PARAMETER,"Invalid parameters, argument \"amount\" must be non-null");
+    
+    if (!metaObj["expiry"].isNull()) makeoffer.expiry = metaObj["expiry"].get_int();
+
+    int targetHeight;
+    {
+        LOCK(cs_main);
+        auto order = pcustomcsview->GetICXOrderByCreationTx(makeoffer.orderTx);
+        if (!order)
+            throw JSONRPCError(RPC_INVALID_PARAMETER, strprintf("orderTx (%s) does not exist",makeoffer.orderTx.GetHex()));
+
+        // if (order->amountToFill * order->orderPrice / COIN < makeoffer.amount)
+        //     throw JSONRPCError(RPC_INVALID_PARAMETER, "cannot make offer with that amount, order (" + order->creationTx.GetHex() + ") has less amount to fill!");
```

Comment:
Why remove this check?

Reply:
I noticed that doing double checks in both rpc and mn_checks can lead to some bugs, because if the check is slightly different you can miss something. For example if you check x < 4 in rpc and x <= 4 in mn checks you won't be able to send tx with x == 4. This is not a big issue because if mn_checks is last check point and we can change RPC without consensus change. But I realized that I can miss some checks if I do it on both sides. So basically whatever is checked in mn_checks can be removed from RPC...